internal/provider/resource_vpc_firewall_rules.go

```diff
@@ -388,19 +383,14 @@ func (r *vpcFirewallRulesResource) Create(ctx context.Context, req resource.Crea
 	// This means we'll set it here solely for Terraform.
 	plan.ID = types.StringValue(uuid.New().String())
 
-	// The order of the response is not guaranteed to be the same as the one set
-	// by the tf files. We will be populating all values, not just computed ones
-	plan.Rules, diags = newVPCFirewallRulesModel(firewallRules.Rules)
-	resp.Diagnostics.Append(diags...)
-	if resp.Diagnostics.HasError() {
-		return
-	}
-
 	plan.TimeCreated = types.StringNull()
 	plan.TimeModified = types.StringNull()
 	if len(plan.Rules) > 0 {
-		plan.TimeCreated = plan.Rules[0].TimeCreated
-		plan.TimeModified = plan.Rules[0].TimeModified
+		for _, rule := range plan.Rules {
+			plan.TimeCreated = rule.TimeCreated
+			plan.TimeModified = rule.TimeModified
+			break
+		}
```

Comment:
We only need `time_created` and `time_modified` values from any rule. Perhaps we can extract this into a helper function since we're doing this in a few places now.